metric/prometheus.go

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2022, MegaEase
+ * All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package metric
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	log "github.com/sirupsen/logrus"
+)
+
+const module = "Metric"
+
+// MetricsType is the generic type of metrics
+type MetricsType interface {
+	*prometheus.CounterVec | *prometheus.GaugeVec | *prometheus.HistogramVec | *prometheus.SummaryVec
+}
+
+var counterMap = make(map[string]*prometheus.CounterVec)
```

Comment:
four `maps define` include `four types` and belong to a set of definitions. 
For readability, it is more recommended to:
```
var (
    counterMap   = make(map[string]*prometheus.CounterVec)
    gaugeMap     = make(map[string]*prometheus.GaugeVec)
    histogramMap = make(map[string]*prometheus.HistogramVec)
    summaryMap   = make(map[string]*prometheus.SummaryVec)
)
```